xbmc/interfaces/json-rpc/AudioLibrary.cpp

```diff
@@ -1362,25 +1360,20 @@ JSONRPC_STATUS CAudioLibrary::RefreshArtist(const std::string& method,
                                             const CVariant& parameterObject,
                                             CVariant& result)
 {
-  int artistID = (int)parameterObject["artistid"].asInteger();
-
-  CMusicDbUrl musicUrl;
-  if (!musicUrl.FromString("musicdb://artists/"))
-    return InternalError;
-
   CMusicDatabase musicdatabase;
   if (!musicdatabase.Open())
     return InternalError;
 
-  //checking if artistID is a valid one
+  // Checking if artistID is a valid one
+  int artistID = (int)parameterObject["artistid"].asInteger();
   if (!musicdatabase.GetArtistExists(artistID))
     return InvalidParams;
 
-  //set the artist id on the musicdb url
-  musicUrl.AddOption("artistid", artistID);
-
-  //executing the StartArtistScan for refreshing the artist scraped information
-  CMusicLibraryQueue::GetInstance().StartArtistScan(musicUrl.ToString(), true);
+  // Start rescraping additional information for the given artist
+  const std::string cmd =
+      StringUtils::Format("musiclibrary.refreshartist({})",
+                          StringUtils::Paramify(parameterObject["artistid"].asString()));
```

Comment:
For performance reasons I would introduce a variable for the artistid variant and use it as the two place where you need it in this method.

`const CVariant artistIdVariant{parameterObject["artistid"]};`